src/viur/core/skeleton/tasks.py

```diff
@@ -129,151 +141,150 @@ def __txn_update(_skel, key, srcRelKey):
-            continue
-        if not (src_prop := relation_object.get("viur_src_property")):
-            logging.critical("We got an relation-object without a src_prop!")
-            continue
-        try:
-            skel = skeletonByKind(src_kind)()
-        except AssertionError:
-            # The referenced skeleton does not exist in this data model -> drop that relation object
-            logging.info(f"Deleting {relation_object.key} which refers to unknown kind {src_kind}")
-            db.delete(relation_object)
-            count_removed += 1
-            continue
-        if src_prop not in skel:
-            logging.info(f"Deleting {relation_object.key} which refers to "
-                         f"non-existing RelationalBone {src_prop} of {src_kind}")
-            db.delete(relation_object)
-            count_removed += 1
-    logging.info(f"END processVacuumRelationsChunk {module}, "
-                 f"{count_total} records processed, {count_removed} removed")
-    if new_cursor := query.getCursor():
-        # Start processing of the next chunk
-        processVacuumRelationsChunk(module, new_cursor, count_total, count_removed, notify)
-    elif notify:
-        txt = (
-            f"{conf.instance.project_id}: Vacuum relations finished for {module}\n\n"
-            f"ViUR finished to vacuum viur-relations for module {module}.\n"
-            f"{count_total} records processed, "
-            f"{count_removed} entries removed"
+        class FilterRowUsingSkel(RelSkel):
+            name = StringBone(
+                required=True,
+            )
+
+            op = SelectBone(
+                required=True,
+                values={
+                    "$eq": "=",
+                    "$lt": "<",
+                    "$gt": ">",
+                    "$lk": "like",
+                },
+                defaultValue=" ",
+            )
+
+            value = StringBone(
+                required=True,
+            )
+
+        filters = RecordBone(
+            descr="Filter",
+            using=FilterRowUsingSkel,
+            multiple=True,
+            format="$(name)$(op)=$(value)",
         )
+
+        condition = RawBone(
+            descr="Condition",
+            required=True,
+            defaultValue="False  # fused: by default, doesn't affect anything.\n",
+            params={
+                "tooltip": "Enter a Logics expression here to filter entries by specific skeleton values."
+            },
+        )
+
+    def execute(self, task, kinds, filters, condition):
         try:
-            email.send_email(dests=notify, stringTemplate=txt, skel=None)
-        except Exception as exc:  # noqa; OverQuota, whatever
-            logging.exception(f"Failed to notify {notify}")
+            logics.Logics(condition)
+        except logics.ParseException as e:
+            raise errors.BadRequest(f"Error parsing condition {e}")
+
+        notify = current.user.get()["name"]
+
+        for kind in kinds:
```

Comment:
What if I want to run a task for all kinds the I musst select all and send it ? Can we use a wildcard here ?

Reply:
Good Idea i Take a Look in that tomorrow